### PR TITLE
Python syntax method calls

### PR DESCRIPTION
**Pull request by 1mn**
image

image

### CODE REVIEW

**Comment by gdude2002** [comment]
image

```
Traceback (most recent call last):
  File "C:\Python36\lib\site-packages\discord\client.py", line 223, in _run_event
    yield from coro(*args, **kwargs)
  File "D:\Programming\discord-python\bot\bot\cogs\verification.py", line 21, in on_message
    ctx = await self.bot.get_context(message)  # type: Context
  File "C:\Python36\lib\site-packages\discord\ext\commands\bot.py", line 864, in get_context
    invoker = view.get_word()
  File "D:\Programming\discord-python\bot\bot\__init__.py", line 48, in case_insensitive_get_word
    new_args[-1] = new_args[-1].replace(")", "")
IndexError: list index out of range
```

image

image

**Comment by gdude2002** [comment]
This seems to work quite well, but that exception should probably be replaced with the command's help text or something

File: bot/__init__.py [hunk 1]
```diff
@@ -23,18 +25,48 @@ def case_insensitive_get_word(self) -> str:
...
 
+    if isinstance(result, str):
+        return result.lower()  # Case insensitivity, baby
+    return result
 
-# Save the old methods
-_skip_string = discord.ext.commands.view.StringView.skip_string
-_get_word = discord.ext.commands.view.StringView.get_word
 
 # Monkey patch them to be case insensitive
```
Comment (gdude2002) [inline bot/__init__.py]: Incorrect comment I guess

**Comment by 1mn** [comment]
This one wasn't easy.
image

File: bot/__init__.py [hunk 2]
```diff
@@ -18,24 +20,84 @@ def case_insensitive_skip_string(self, string: str) -> bool:
...
+        first = double_quotes
+        second = single_quotes
+
+    # after we get the first arg, we remove it from the buf so that
+    # quotes inside the arg will not be picked up in the second findall.
+    new_args = re.findall(first, buf)
+    print(new_args)
+    for arg in new_args:
+        buf = buf.replace(arg, "")
+    new_args += re.findall(second, buf)
```
Comment (AkiraSama) [inline bot/__init__.py]: In the process of getting myself a staging setup of the bot, but for others' reference while I'm working on that:

This looks like it'll turn the arguments from
```python
bot.command_with_four_args("first", 'second', "third", 'fourth')
```
into
```python
["first", "third", "second", "fourth"]
```

Comment (1mn) [reply bot/__init__.py]: by george. you're right. of course you're right. yeah I need to fix that.

Comment (AkiraSama) [reply bot/__init__.py]: If that's the case, requesting changes to remove approval. Not sure if we're keeping the debug `print` statements or not but that's another thing I had on the table.

Comment (1mn) [reply bot/__init__.py]: dismissing approval. fuck it, I guess we're gonna try with `ast` after all.

**Comment by AkiraSama** [comment]
Looks good for our purposes atm

**Pull request merged**